resolver/resolver.cc

```diff
@@ -1452,6 +1553,21 @@ class ResolveConstantsWalk {
             return;
         }
 
+        // Populate local first definitions table for out-of-order reference checking.
+        // This needs to happen *iff* the symbol is:
+        //   a). defined only in this file and nowhere else
+        //   b). defined multiple times in this file
+        //   c). this particular assign is the first occurrence of the symbol in the file
+        if (loadScopeDepth_ == 0 && id->symbol.isOnlyDefinedInFile(ctx.state, ctx.file)) {
```

Comment:
Same question here about RBI files.  And, actually, since this code and the above are so similar and require the same amount of commentary, maybe there should be a single function `(ctx, sym, declLoc)` that handles all this logic?